robottelo/decorators/func_shared/file_storage.py

```diff
@@ -0,0 +1,100 @@
+# -*- encoding: utf-8 -*-
+import logging
+import os
+import tempfile
+
+from pytest_services.locks import file_lock
+
+from robottelo.decorators.func_shared.base import (
+    BaseStorageHandler,
+    DEFAULT_LOCK_TIMEOUT
+)
+
+TEMP_ROOT_DIR = 'robottelo'
+TEMP_FUNC_SHARED_DIR = 'shared_functions'
+SHARED_DIR = None
+
+logger = logging.getLogger(__name__)
+
+
+def _get_root_dir(create=True):
+    global SHARED_DIR
+    if SHARED_DIR is not None:
+        return SHARED_DIR
+
+    tmp_root_dir = os.path.join(tempfile.gettempdir(), TEMP_ROOT_DIR,
+                                TEMP_FUNC_SHARED_DIR)
+    if create and not os.path.exists(tmp_root_dir):
+        try:
+            # it can happen that the workers try to create this path at the
+            # same time
+            os.makedirs(tmp_root_dir)
+        except OSError:
+            if not os.path.exists(tmp_root_dir):
+                raise
+
+    SHARED_DIR = tmp_root_dir
+
+    return SHARED_DIR
+
+
+class FileStorageHandler(BaseStorageHandler):
+    """Key value file storage handler."""
+
+    def __init__(self, root_dir=None, create=True,
+                 lock_timeout=DEFAULT_LOCK_TIMEOUT):
+
+        if root_dir is None:
+            root_dir = _get_root_dir()
+
+        if create and not os.path.exists(root_dir):
+            os.makedirs(root_dir)
+
+        self._lock_timeout = lock_timeout
+        self._root_dir = root_dir
+
+    @property
+    def root_dir(self):
+        return self.root_dir()
+
+    def get_key_file_path(self, key):
+        return os.path.join(self._root_dir, key)
+
+    def lock(self, key):
+        lock_key = '{}.lock'.format(key)
+        return file_lock(self.get_key_file_path(lock_key), remove=False,
+                         timeout=self._lock_timeout)
+
+    def when_lock_acquired(self, handler):
```

Comment:
1 - I like the pickle functionality, but we need a more readable and basic to handle format for debugging in case of trouble 
2 - shelve here is useless as any way we need to implement a locking mechanism over keys
3 - if I had to choose a python pickle mechanism I much prefer zodb, that have transactions and a locking mechaniz can be easily implemented https://github.com/zopefoundation/ZODB , but for multi processes usage we have to start the ZEO service.